cargo-registry/src/crate_handler.rs

```diff
@@ -159,20 +199,23 @@ impl From<&UnpackedCrate> for Program {
             path: value.program_path.clone(),
             id: value.program_id,
             _tempdir: value.tempdir.clone(),
+            meta: value.meta.clone(),
+            crate_bytes: value.crate_bytes.clone(),
         }
     }
 }
 
-pub(crate) struct CratePackage(pub(crate) Bytes);
+#[derive(Clone, Default)]
+pub(crate) struct CrateTarGz(pub(crate) Bytes);
 
-impl From<UnpackedCrate> for Result<CratePackage, Error> {
+impl From<UnpackedCrate> for Result<CrateTarGz, Error> {
```

Comment:
Isn't this TryFrom? But also maybe it should just be an utility method?

From and TryFrom are supposed to be for cheap conversions, this is compressing so doesn't seem cheap. The rationale for having From and TryFrom be cheap is that for example if you're reading the code and you find a something.into() you don't expect it to potentially be a perf sensitive thing. Instead if you see `return compress_data(data)` you immediately know it might be an heavy operation.

Reply:
That's a good point. I'll update the code.